### ISSUE TITLE
undefined is not an object (evaluating 'args.length')

### ISSUE THREAD

**Issue by vinnitu** [marcelduran/yslow#129]
for phantomjs 2.0.0dev

phantomjs yslow.js --help

output is

```
TypeError: undefined is not an object (evaluating 'args.length')

  yslow.js:5 in global code
```

**Comment by lewsid** [comment]
Seeing the same thing, using PhantomJS 2.0. Looks like yslow can no longer access the phantomJS arguments.

**Comment by lewsid** [comment]
Digging in a little, and found that 'phantom.args' is deprecated in v2 (http://phantomjs.org/api/phantom/property/args.html). I tried updating the relevant code to use the newer 'system.args' call and it no longer throws an error, but it also no longer finishes.

![screen shot 2015-01-29 at 10 11 18 am](https://cloud.githubusercontent.com/assets/163191/5955073/3171279a-a79f-11e4-80e1-b2fc68fda2cc.png)

Bummer. Looks like I'll have to dig a little deeper.

**Comment by soulgalore** [comment]
yep, the  <code>phantom.args</code> has been removed in 2.0, I've changed that in https://github.com/sitespeedio/yslow/blob/master/src/phantomjs/controller.js using <code>system.args</code> instead (thanks @thebostik for showing me). You can cherry pick the fix from there.

**Comment by soulgalore** [comment]
@lewsid the first argument in system.args is the script name so you need to adjust the arguments, then it will work.

**Comment by lewsid** [comment]
Thanks for the info @soulgalore! I've applied your fix, and rearranged my arguments, but it still hangs. My command looks like this:

`/usr/bin/phantomjs_2.0 /path/to/patched_yslow.js --ssl-protocol=any --info basic http://example.com`

Can you spot anything that I'm doing obviously wrong here?

**Comment by soulgalore** [comment]
@lewsid if you could share your branch, then I can try it out, and then you can make a PR so it's fixed in YSlow?

**Comment by jpkeisala** [comment]
@lewsid Did you manage to to get yslow working? I am also looking the fix for this.

**Comment by lewsid** [comment]
@jpkeisala Unfortunately, no. Sorry to leave you and others hanging without my code for reference. I made some progress but eventually abandoned the effort and just nixed yslow from the project.

**Comment by soulgalore** [comment]
I can make the PR tonight then, np.

**Comment by abcfy2** [comment]
Did this issue fix for now? I'm facing the same issue.

I'm using the latest phantomjs(as of `2.1.1`), yslow from http://yslow.org/yslow-phantomjs-3.1.8.zip, and raise an error:

```
TypeError: undefined is not an object (evaluating 'args.length')

  phantomjs://code/yslow.js:5 in global code
```

**Comment by soulgalore** [comment]
@abcfy2 you can use [this PR](https://github.com/marcelduran/yslow/pull/150) to fix it.

**Comment by abcfy2** [comment]
@soulgalore Thanks, it works.

But do you know how to use yslow for login pages ? I know phantomjs can work as a real browser, but I don't know how to implement.

**Comment by Muthulakshmirp** [comment]
I'm facing the same issue as well with version 3.1.8

Could anyone help?

TypeError: undefined is not an object (evaluating 'args.length')

  phantomjs://code/yslow.js:5 in global code

**Comment by paramesh** [comment]
To make it much simpler to follow just do the following.

1.) Open yslow.js file under yslow-phantomjs-3.1.8
2.) Replace the following line
var i,arg,page,urlCount,viewport,webpage=require("webpage"),args=phantom.args,
with 
var i,arg,page,urlCount,viewport,webpage=require("webpage"),system = require('system'),args=system.args,

The above works for me.

**Comment by davidfavor** [comment]
Problem still exists in yslow-phantomjs-3.1.8 + the above fix is difficult because yslow.js seem obfuscated.

Be great if this problem could be fixed.

Also, fix show above does not work.

Changes the error to...
```
TypeError: undefined is not an object (evaluating 'args.length')
```
